tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -516,11 +516,12 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     //   bitcast(add(gemm(a, b), bitcast(broadcast(bias)))) ->
     //   bitcast(gemm(a, b, bitcast(broadcast(bias)))) (FuseMatrixBiasAdd)
     //
-    if (Match(instr,
-              m::AddAnyOrder(
-                  m::Bitcast(CublasLtMatmul(&existing_gemm).WithOneUser())
-                      .WithOneUser(),
-                  m::Broadcast(&bias, m::Op()).WithOneUser()))) {
+    if (Match(
+            instr,
+            m::AddAnyOrder(
+                m::Bitcast(CublasLtMatmulMaybeF8(&existing_gemm).WithOneUser())
```

Comment:
This optimization is only valid if the bitcast does not change the shape, and is only valid for certain layouts. I haven't thought through what layouts this optimization is valid for, but it's valid if all layouts have the dimensions ordered from most major to most minor.

CC @philipphack

Reply:
The `bitcast` stands there even for non-fp8. So should fp8 be treated differently here?

Reply:
I think this should be fine if the bitcast acts as a reshape.

Reply:
@reedwm Please get back to wenscarl and Philipp regarding the change you requested here.

Reply:
I couldn't find a case where a bitcast did not act a reshape. I am ok with assuming it acts as a reshape for now.